src/loaders/express/helmet.ts

```diff
@@ -73,10 +74,22 @@ const helmetMiddlewares = () => {
         'https://www.gstatic.cn/',
       ],
       formAction: ["'self'"],
-      upgradeInsecureRequests: !config.isDev,
-      reportUri: config.cspReportUri,
     },
-  })
+  }
+
+  // Add on upgradeInsecureRequest CSP header if !config.isDev
+  if (!config.isDev) {
+    set(cspDirectives, 'directives.upgradeInsecureRequests', [])
```

Comment:
tiny quibble: I would actually just do `cspDirectives.directives.upgradeInsecureRequests = []` here, and something similar for `cspReportUri`. We don't want these lines of code to fail silently if someone accidentally messes up the options.

Reply:
to clarify, you mean without the conditional if (!config.isDev)? or to not use lodash set

Reply:
as in I think it's better not to use lodash set

Reply:
edited